src/decimal128.rs

```diff
@@ -23,6 +27,8 @@ pub struct Decimal128 {
 }
 
 #[cfg(feature = "decimal128")]
+#[deprecated = "The feature-gated decimal128 implementation serializes to BSON incorrectly and \
+                should not be used. It will be removed completely in 2.0.0"]
 impl Decimal128 {
```

Comment:
To only deprecate the feature-flag and not the `Decimal128` struct itself, I only added the attribute to this impl block.